File: libs/src/java/salmon-core/com/mku/streams/Buffer.java
package com.mku.streams;

import java.util.Arrays;

/**
 * Buffer that can be used for buffered streams.
 */
public class Buffer {
    public byte[] getData() {
        return data;
    }

    /**
     * Set the data
     *
     * @param data The data
     */
    public void setData(byte[] data) {
        this.data = data;
    }

	/**
     * Get the start position
     *
     * @return The start position
     */
    public long getStartPos() {
        return startPos;
    }

    /**
     * Set the start position
     *
     * @param startPos The start position
     */
    public void setStartPos(long startPos) {
        this.startPos = startPos;
    }

	/**
     * Get the data count
     *
     * @return The data count
     */
    public long getCount() {
        return count;
    }

    /**
     * Set the data count
     *
     * @param count The data count
     */
    public void setCount(long count) {
        this.count = count;
    }

    private byte[] data;
    private long startPos = 0;
    private long count = 0;

    /**
     * Instantiate a buffer.
     *
     * @param bufferSize The buffer size
     */
    public Buffer(int bufferSize) {
        data = new byte[bufferSize];
    }

    /**
     * Clear the buffer.
     */
    public void clear() {
        if (data != null)
            Arrays.fill(data, 0, data.length, (byte) 0);
    }
}

File: libs/src/csharp/SalmonCore/Streams/Buffer.cs
/*
MIT License

Copyright (c) 2021 Max Kas

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
SOFTWARE.
*/

using System;

namespace Mku.Streams;


/// <summary>
/// Buffer that can be used for buffered streams.
/// </summary>
public class Buffer
{
    /// <summary>
    /// The buffer
    /// </summary>
    public byte[] Data { get; set; }

    /// <summary>
    /// The starting position
    /// </summary>
    public long StartPos { get; set; } = 0;

    /// <summary>
    ///  The count of bytes used
    /// </summary>
    public long Count { get; set;} = 0;

    /// <summary>
    ///  Instantiate a cache buffer.
    /// </summary>
    ///  <param name="bufferSize">The buffer size</param>
    public Buffer(int bufferSize)
    {
        Data = new byte[bufferSize];
    }

    /// <summary>
    ///  Clear the buffer.
    /// </summary>
    public void Clear()
    {
        if (Data != null)
            Array.Fill<byte>(Data, 0, Data.Length, (byte)0);
    }
}